### PR TITLE
server: fix a regression introduced in #186

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Fixes a regression caused in #186. The `event.Issue == nil` check was avoiding pr to stop falling through.

### CODE REVIEW

File: server/issue_handler.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ func (s *Server) issueEventHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
+	defer r.Body.Close()
```
Comment (agnivade) [inline server/issue_handler.go:36]: This seems unrelated?

Comment (isacikgoz) [reply server/issue_handler.go:36]: Not specifically related with the check but it's an error haven't been discovered yet :)

Comment (agnivade) [reply server/issue_handler.go:36]: I don't believe we need to close the body in server response handlers. This is only applicable for clients making the requests.

> // The Server will close the request body. The ServeHTTP
    // Handler does not need to.

File: server/github_pr.go [hunk 2]
```diff
@@ -61,18 +60,14 @@ func (d *EventData) HasUpdateBranch() bool {
...
 
-func PullRequestEventFromJSON(data io.Reader) (*PullRequestEvent, error) {
+func PullRequestEventFromJSON(data io.Reader) *PullRequestEvent {
 	decoder := json.NewDecoder(data)
 	var event PullRequestEvent
 	if err := decoder.Decode(&event); err != nil {
-		return nil, err
-	}
-
-	if event.Issue == nil {
```
Comment (khos2ow) [inline server/github_pr.go:71]: I don't see why removing this would have any effect. If `event.Isssue` is nil then the `err` would be non-nil so `if err == nil && event.PRNumber != 0 {` block would never get executed.

File: server/issue_handler.go (hunk 1 above)
Comment (isacikgoz) [reply server/issue_handler.go:36]: I see, didn't know that. Thanks 👍

File: server/github_pr.go (hunk 2 above)
Comment (agnivade) [reply server/github_pr.go:71]: Not sure I understand your point completely. If event.Issue is nil, we were returning an error previously, which prevented the `if err == nil && event.PRNumber != 0 {` block from getting executed. Now it will.

Perhaps I am missing something?

Comment (khos2ow) [reply server/github_pr.go:71]: I have the impression if `event.Issue` is not nil then `event.PRNumber` is `0`. Isn't this the case?

Comment (agnivade) [reply server/github_pr.go:71]: Not really. It's not mutually exclusive. The `Issue` field might be there for both scenarios. But the `PRNumber` will only be there if it's a PR.

It's super confusing, and that's why we are moving to separate endpoints.

Comment (khos2ow) [reply server/github_pr.go:71]: Oh I see, it makes sense then!
